src/controllers/authenticationController.ts

```diff
@@ -105,9 +155,62 @@ export default class AuthenticationController {
       .send(metadata);
   }
 
-  private return500Error(error: Error, res: express.Response): void {
-    res.status(500).json({
-      message: error.message
+  /**
+   * Wrap the spidStrategy.logout function in a new Promise.
+   */
+  private spidLogout(req: express.Request): Promise<Either<Error, string>> {
+    return new Promise(resolve => {
+      this.spidStrategy.logout(req, (err, request: express.Request) => {
+        if (!err) {
+          return resolve(
+            right<Error, string>(
+              JSON.stringify({
+                logoutUrl: request
+              })
+            )
+          );
+        } else {
+          return resolve(left<Error, string>(err));
+        }
+      });
     });
   }
+
+  /**
+   * Extracts the Bearer token from the authorization header in the request.
+   */
+  private extractTokenFromRequest(
+    req: express.Request
+  ): Promise<Option<string>> {
+    return new Promise(resolve => {
+      if (
+        req.headers &&
+        req.headers.authorization &&
+        typeof req.headers.authorization === "string"
+      ) {
+        const authorization = req.headers.authorization;
+        const parts = authorization.split(" ");
+        if (parts.length === 2) {
+          const scheme = parts[0];
+          const token = parts[1];
+
+          if (/^Bearer$/i.test(scheme)) {
+            resolve(some(token));
+          } else {
+            resolve(none);
+          }
+        }
+      }
+      resolve(none);
+    });
+  }
+
+  /**
+   * Refresh a user token in the session storage and returns the new token.
+   */
+  private async refreshUserToken(
+    token: string
+  ): Promise<Either<Error, ISessionState>> {
+    return await this.sessionStorage.refresh(token);
```

Comment:
what's the point of this function? can't you just call `this.sessionStorage.refresh` ?